x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LocalLogicalPlanOptimizerTests.java

```diff
@@ -744,6 +744,14 @@ public void testUnionTypesInferNonNullAggConstraint() {
         assertEquals("integer_long_field", unionTypeField.fieldName().string());
     }
 
+    public void testGroupingByMissingFields() {
+        var plan = plan("FROM test | STATS AVG(salary) BY first_name, last_name ");
+        var testStats = statsForMissingField("first_name", "last_name");
+        var localPlan = localPlan(plan, testStats);
+        Aggregate aggregate = (Aggregate) localPlan.collectFirstChildren(p -> p instanceof Aggregate).getFirst();
+        assertThat(aggregate.groupings(), hasSize(2));
```

Comment:
I'd expand the test a little to assert the rest of the plan.

With the alternative solution I pushed, there should be an `Eval` upstream taking care of creating a new attribute for the duplication that otherwise happens when the groupings would get optimized away.